File: frontend/src/app/core/models/evenement.ts
export interface Evenement {
  id: number;
  titre: string;              // was: title ✓ fix this
  description?: string;
  dateDebut: string;          // was: date — backend has dateDebut + dateFin
  dateFin?: string;           // missing entirely
  lieu?: string;              // was: location ✓ fix this
  nbParticipantsMax: number;  // was: capacity ✓ fix this
  statut: 'EN_ATTENTE' | 'OUVERT' | 'PLEIN' | 'ANNULE' | 'TERMINE' | 'REJETE'; // was wrong values
  organisateur?: { id: number; nom: string; prenom: string; }; // was: organizerId (it's a nested object, not an ID)
  imageUrl?: string;
  club?: string;
  commentaireAdmin?: string;
  registeredCount?: number; 
}

export interface EventRegistration {
  id?: number;
  etudiantId: number;
  evenementId: number;
}

export interface EventApplication {
  id: number;
  evenementId: number;
  etudiantId: number;
  statut: 'EN_ATTENTE' | 'APPROUVE' | 'REJETE';
  dateDemande?: string;
  commentaireAdmin?: string;
}

export enum EventStatus {
  PENDING = 'EN_ATTENTE',
  APPROVED = 'APPROUVE',
  REJECTED = 'REJETE'
}

File: backend/campushub/src/main/java/tn/enicarthage/campushub/enseignant/model/Evenement.java
package tn.enicarthage.campushub.enseignant.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import tn.enicarthage.campushub.shared.model.User;

@Entity
@Table(name = "evenements")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Evenement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String titre;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private LocalDateTime dateDebut;

    @Column(nullable = false)
    private LocalDateTime dateFin;

    private String lieu;

    private Integer nbParticipantsMax;

    @Enumerated(EnumType.STRING)
    private Statut statut = Statut.EN_ATTENTE;  // default to pending

    @ManyToOne
    @JoinColumn(name = "organisateur_id")
    private User organisateur;

    private String imageUrl;
    private String club;

    @Column(name = "commentaire_admin", length = 1000)
    private String commentaireAdmin;

    private LocalDateTime createdAt = LocalDateTime.now();

    public enum Statut {
        EN_ATTENTE,  // pending approval
        OUVERT,      // approved and open
        PLEIN,       // full
        ANNULE,
        TERMINE,
        REJETE
    }
}